torchsummary/tests/unit_tests/torchsummary_test.py

```diff
@@ -28,9 +31,61 @@ def test_single_layer_network(self):
 
     def test_single_layer_network_on_gpu(self):
         model = torch.nn.Linear(2, 5)
-        model.cuda()
+        if torch.cuda.is_available():
+            model.cuda()
+        input = (1, 2)
+        total_params, trainable_params = summary(model, input, device=gpu_if_available)
+        self.assertEqual(total_params, 15)
+        self.assertEqual(trainable_params, 15)
+
+    def test_multiple_input_types(self):
+        model = MultipleInputNetDifferentDtypes()
+        input1 = (1, 300)
+        input2 = (1, 300)
+        dtypes = [torch.FloatTensor, torch.LongTensor]
+        total_params, trainable_params = summary(
+            model, [input1, input2], device="cpu", dtypes=dtypes)
+        self.assertEqual(total_params, 31120)
+        self.assertEqual(trainable_params, 31120)
+
+
+class torchsummarystringTests(unittest.TestCase):
```

Comment:
I added same kinds of unit tests for summary_string too, (although it looks having too much duplicates..), just checking the type of the result string. Any suggests for enhancement would be thankful! :) @Naireen

Reply:
<img width="648" alt="Screen Shot 2020-01-11 at 3 16 23 PM" src="https://user-images.githubusercontent.com/17661005/72199848-63ed1d00-3485-11ea-895b-489e76eaf0eb.png">

And tests all pass! (attached for certification)

Reply:
These tests are a good start. I agree, there is a lot of duplication. For sufficient testing, I would say that you only actually need one of these tests. You don't need to duplicate all of the previous tests and ensure that it works for all cases, since if it works for one, there really isn't any reason that it shouldn't work for the others, as all you're doing is asserting the presence of the extra return string. The test `test_single_input` is enough.